modules/dnn/src/model.cpp

```diff
@@ -401,4 +407,310 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+            for (int j = 0; j < 4; j++) {
+                approx.push_back(Point((int)(vertex[3-j].x), (int)(vertex[3-j].y)));
+            }
+        } else {
+            double epsilon = arcLength(contour, true) * 0.01;
+            approxPolyDP(contour, approx, epsilon, true);
+            if (approx.size() < 4) continue;
+        }
+        std::vector<Point> polygon;
+        unclip(approx, polygon, unclipRatio, scaleWidth, scaleHeight);
+        results.push_back(polygon);
+    }
+}
+
+void TextDetectionModel::detect(InputArray frame, const float& confThreshold, const float& nmsThreshold,
+                                const std::vector<String>& outNames, CV_OUT std::vector<std::vector<Point>>& results)
+{
+    results.clear();
+
+    std::vector<Mat> outs;
+    impl->predict(*this, frame.getMat(), outs, outNames);
+    CV_Assert(outs.size() == 2);
+    Mat scoreMap = outs[0];
+    Mat geometry = outs[1];
+
+    CV_Assert(scoreMap.dims == 4); CV_Assert(geometry.dims == 4); CV_Assert(scoreMap.size[0] == 1);
+    CV_Assert(geometry.size[0] == 1); CV_Assert(scoreMap.size[1] == 1); CV_Assert(geometry.size[1] == 5);
+    CV_Assert(scoreMap.size[2] == geometry.size[2]); CV_Assert(scoreMap.size[3] == geometry.size[3]);
+
+    std::vector<RotatedRect> boxes;
+    std::vector<float> scores;
+    const int height = scoreMap.size[2];
+    const int width = scoreMap.size[3];
+    for (int y = 0; y < height; ++y)
+    {
+        const float* scoresData = scoreMap.ptr<float>(0, 0, y);
+        const float* x0_data = geometry.ptr<float>(0, 0, y);
+        const float* x1_data = geometry.ptr<float>(0, 1, y);
+        const float* x2_data = geometry.ptr<float>(0, 2, y);
+        const float* x3_data = geometry.ptr<float>(0, 3, y);
+        const float* anglesData = geometry.ptr<float>(0, 4, y);
+        for (int x = 0; x < width; ++x)
+        {
+            float score = scoresData[x];
+            if (score < confThreshold)
+                continue;
+
+            float offsetX = x * 4.0f, offsetY = y * 4.0f;
+            float angle = anglesData[x];
+            float cosA = std::cos(angle);
+            float sinA = std::sin(angle);
+            float h = x0_data[x] + x2_data[x];
+            float w = x1_data[x] + x3_data[x];
+
+            Point2f offset(offsetX + cosA * x1_data[x] + sinA * x2_data[x],
+                           offsetY - sinA * x1_data[x] + cosA * x2_data[x]);
+            Point2f p1 = Point2f(-sinA * h, -cosA * h) + offset;
+            Point2f p3 = Point2f(-cosA * w, sinA * w) + offset;
+            boxes.push_back(RotatedRect(0.5f * (p1 + p3), Size2f(w, h), -angle * 180.0f / (float)CV_PI));
+            scores.push_back(score);
+        }
+    }
+
+    // Apply non-maximum suppression procedure.
+    std::vector<int> indices;
+    NMSBoxes(boxes, scores, confThreshold, nmsThreshold, indices);
+
+    // Re-scale
+    Point2f ratio((float)frame.cols() / impl->size.width, (float)frame.rows() / impl->size.width);
```

Comment:
shouldn't it be:

    (float)frame.rows() / impl->size.height;

Reply:
Yes, I am sorry about it. The input width and height are the same, so I did not find it.